lib/Infer/EnumerativeSynthesis.cpp

```diff
@@ -751,70 +812,40 @@ void generateAndSortGuesses(SynthesisContext &SC,
     PruneFuncs.push_back(DataflowPruning.getPruneFunc());
   }
   auto PruneCallback = MkPruneFunc(PruneFuncs);
-  // TODO(zhengyangl): Refactor the syntactic pruning into a
-  // prune function here, between Cost and Dataflow
-  // TODO(manasij7479) : If RHS is concrete, evaluate both sides
-  // TODO(regehr?) : Solver assisted pruning (should be the last component)
 
-  getGuesses(Guesses, Cands, SC.LHS->Width,
-             LHSCost, SC.IC, nullptr, nullptr, TooExpensive, PruneCallback);
-  if (DebugLevel >= 1) {
-    DataflowPruning.printStats(llvm::errs());
-  }
+
+  std::vector<Inst *> Guesses;
+  uint64_t GuessCount = 0;
+
+  auto Generate = [&SC, &Guesses, &RHS, &EC, &GuessCount](Inst *Guess) {
```

Comment:
Instead of wrapping the function into lambda with complex capture and passing it around, why not just use a normal function, that would make code much more readable :)

Reply:
That won't work because of the captures. One solution is making a class with operator () overloaded. Not sure if that's cleaner than this.